ebos/eclwriter.hh

```diff
@@ -419,18 +458,46 @@ private:
                 // TODO (?): use the cartesian index mapper to make this code work
                 // with grids other than Dune::CpGrid. The problem is that we need
                 // the a mapper for the sequential grid, not for the distributed one.
-                int cc1 = globalGrid_.globalCell()[c1];
-                int cc2 = globalGrid_.globalCell()[c2];
+                std::size_t cc1 = globalGrid_.globalCell()[c1];
```

Comment:
nit: this is probably a case where `auto` would be more readable because if it is decided that the return value of `grid.globalCell()[c1]` should change its signeness, no follow up to get rid of the warning will be needed.

Reply:
Well, this is exactly there to prevent such warnings as `globalGrid_.glglobalCell()` returns a `std::vector<int>` and `eclState().getInputGrid().getNX()` a `std::size_t`. I just reckoned that a negative global index would never appear. That is up for debate of course. But it cannot become `auto` without producing warnings. So it is either the new type or the old. I do not care.

Reply:
okay, seems like that API has a lot of potential for improvement...

Reply:
`int` or `std::size_t`? Your call.

Reply:
I don't really care. `int` does reduces the code grunt and is the type stored in the container stored `globalCell()`, so this is IMO slightly better.

Reply:
Mmh, then I need to change `auto cellDiff` to ìnt cellDiff, too. Otherwise I will get
```
dune/opm/opm-simulator/ebos/eclwriter.hh:399:43: warning: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int’ [-Wsign-compare]
             if (cellDiff != 1 && cellDiff != nx && cellDiff != nx*ny)
                                  ~~~~~~~~~^~~~~
```
And I would need to do another cast to prevent:
```
src/dune/opm/opm-simulator/ebos/eclwriter.hh:480:76: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
                   while ( candidate != nncData.end() && candidate->cell1 == cc1
                                                         ~~~~~~~~~~~~~~~~~^~~~~~
```
I will stick with the way this is and wonder whether this nitpick was really worth investing any time (and time it did cost!).

Reply:
I clearly stated that it is a nit, and I do not insist on fixing nits before merging. If you wasted time because of this, I don't have an idea what else I should do about this...

Reply:
I did even forget that int = unsigned int is a narrowing conversion.